src/data/providers/balancer-api/modules/pool-state/index.ts

```diff
@@ -0,0 +1,88 @@
+import { BalancerApiClient } from '../../client';
+import { PoolStateInput } from "../../../../../entities";
+
+export class Pools {
+  readonly poolStateQuery = `query GetPool($id: String!){
+    poolGetPool(id:$id) {
+      id
+      address
+      name
+      type
+      version
+      ... on GqlPoolWeighted {
+        tokens {
+          ... on GqlPoolTokenBase {
+            address
+             decimals
+            index
+          }
+        }
+      }
+      ... on GqlPoolStable {
+        tokens {
+          ... on GqlPoolTokenBase {
+            address
+             decimals
+            index
+          }
+        }
+      }
+      ... on GqlPoolPhantomStable {
+        tokens {
+          ... on GqlPoolTokenBase {
+            address
+             decimals
+            index
+          }
+        }
+      }
+      ... on GqlPoolGyro {
+        tokens {
+          ... on GqlPoolTokenBase {
+            address
+             decimals
+            index
+          }
+        }
+      }
+      ... on GqlPoolLiquidityBootstrapping {
+        tokens {
+          ... on GqlPoolTokenBase {
+            address
+             decimals
+            index
+          }
+        }
+      }
+      ... on GqlPoolElement {
+        tokens {
+          ... on GqlPoolTokenBase {
+            address
+             decimals
+            index
+          }
+        }
+      }
+      ... on GqlPoolLiquidityBootstrapping {
+        tokens {
+          ... on GqlPoolTokenBase {
+            address
+             decimals
+            index
+          }
+        }
+      }
+    }
+}`;
+
+  constructor(private readonly balancerApiClient: BalancerApiClient) {}
+
+  async fetchPoolState(id: string): Promise<PoolStateInput> {
+    const {
+      data: { poolGetPool },
+    } = await this.balancerApiClient.fetch('GetPool', this.poolStateQuery, {
```

Comment:
Is the `GetPool` (or even `operationName`) necessary? Can you explain what they are as I don't think I've seen it before.

Reply:
The "operationName" is not mandatory when you have only one function in the query, like we do, I can remove it, it's used to specify which function of the query will be executed